src/components/core/geoChart/GeoChartRenderer.tsx

```diff
@@ -33,16 +34,26 @@ export default class GeoChartRenderer extends React.PureComponent<IGeoChartRende
     private chart: mapboxgl.Map;
     private chartRef: HTMLElement;
 
-    public componentDidMount() {
-        this.createMap();
+    public componentDidUpdate(prevProps: IGeoChartRendererProps) {
+        if (!isEqual(this.props.execution, prevProps.execution)) {
```

Comment:
We should move this check to `shouldComponentUpdate` as above comment

Reply:
fixed